src/Native/LibTorchSharp/Utils.h

```diff
@@ -10,6 +10,8 @@ extern thread_local char *torch_last_err;
 
 typedef torch::Tensor *Tensor;
 typedef torch::Scalar *Scalar;
+typedef torch::Generator* Generator;
```

Comment:
So it would be good to write some notes on when we use `shared_ptr` reference counting and why.

Reply:
That's a fair question. When I put this in place, I followed the Tensor pattern rather than the Module / Optimizer pattern. The generators are already implemented using their own proprietary reference counting scheme, so it seemed like adding a layer of reference counting on top of an already existing one. Redundant.

Happy to be convinced that was misguided.